### PR TITLE
Fixed render_headers accepting nil, blank and empty values

### LINKED ISSUE #60: An empty header field causes an infinite loop on render
Putting an empty or nil value into a header will cause an infinite loop in the `render_header_value`
Since there isn't a function for handling empty input.

**Version**
Commit 99facb9860aade3245c2a3288819fd81da31d0e7

**Steps to Reproduce**

```elixir
mail =
  Mail.build
  |> Mail.Message.put_header("x-mms-message-id", nil)
  |> Mail.render
```

**Test Case**
What test case? This thing just loops like nobody's business, you figure it out.

**Expected Behaviour**
Doesn't infinitely loop.
Jokes aside, it should either raise an error, or discard the header.

**Actual Behaviour**
It loops.

### PR DESCRIPTION
**Pull request by IceDragon200**
FIXED :3

Fixes #60

### CODE REVIEW

File: test/mail/renderers/rfc_2822_test.exs [hunk 1]
```diff
@@ -42,6 +42,11 @@ defmodule Mail.Renderers.RFC2822Test do
     assert headers == "Foo: bar\r\nBaz: qux"
   end
 
+  test "headers - handles empty headers" do
+    headers = Mail.Renderers.RFC2822.render_headers(%{"content-type" => "text/plain", "message-id" => nil, "to" => [], "from" => "      ", "content-disposition" => "attachment"})
```
Comment (bcardarella) [inline test/mail/renderers/rfc_2822_test.exs]: there needs to be more tests here. I see three new function signatures added but only one tests case.

Comment (IceDragon200) [inline test/mail/renderers/rfc_2822_test.exs]: Hmm, I did handle all three cases in that line, but I guess splitting them would be the correct approach

Comment (bcardarella) [inline test/mail/renderers/rfc_2822_test.exs]: oh, I'm sorry I missed the other header values. This works :)

Comment (IceDragon200) [inline test/mail/renderers/rfc_2822_test.exs]: And I've split them into separate tests now

**Comment by IceDragon200** [comment]
@bcardarella  CI tests have finished

**Comment by archSeer** [comment]
Approved but not merged?

**Comment by bcardarella** [comment]
Thanks!

**Pull request merged**